### DISCUSSION TITLE
长按倍速后恢复原速

### DISCUSSION THREAD

**Discussion by WuYoue** [dyphire/mpv-config#152] [General]
请教一下大佬，在长按倍速播放以后松开，倍速会回到1倍速，例如当前设置1.2倍速，长按倍速播放后变成了1倍速，并没有回到原来的1.2倍速，请问这个应该怎么解决呀？

**Comment by dyphire** [comment]
长按释放后恢复原本的播放速度是预期行为，绝大多数播放器的长按倍速功能都是如此，我不认为这有什么问题
如果想要持久化的倍速播放可以通过自行绑定新的倍速命令或通过鼠标调节播放 UI 中间的速度控制旋钮来实现：
![image](https://github.com/user-attachments/assets/7d5802e9-d442-44fc-9c4b-933af8af3d25)

**Comment by WuYoue** [comment]
目前主流视频网站（B站、抖音、腾讯视频等），设置1.5倍速后，长按3倍速播放，在松开后会回到1.5倍速播放。并不是设置1.5倍数后，松开仍然回到1倍速，如图所示。这样可以在听网课时加速跳过无聊的片段，又不错过期间可能出现的重要知识点。希望大佬能实现该功能，谢谢啦。💕💕💕
![b1](https://github.com/user-attachments/assets/50ac5531-f735-4d28-8f52-78a663f30251)
![b2](https://github.com/user-attachments/assets/72b647f2-d396-45f2-8bc6-78dbc02b08f0)
![b3](https://github.com/user-attachments/assets/45a3f529-7b7e-4ced-8c0f-95416ec15459)

**Comment by verygoodlee** [comment]
这个仓库里的evafast应该还是比较旧的master分支
https://github.com/dyphire/mpv-config/blob/32464e21f5d91c88d60ea35af27f1c511b7fd287/manager.json#L2-L6
rewrite分支早就实现了恢复原速的功能，可自行替换`evafast.lua` `evafast.conf`试试
https://github.com/po5/evafast/tree/rewrite

**Reply by dyphire** [reply]
> 这个仓库里的evafast应该还是比较旧的master分支
> 
> https://github.com/dyphire/mpv-config/blob/32464e21f5d91c88d60ea35af27f1c511b7fd287/manager.json#L2-L6
> 
> 
> rewrite分支早就实现了恢复原速的功能，可自行替换`evafast.lua` `evafast.conf`试试
> https://github.com/po5/evafast/tree/rewrite

应该还有 inputevent 中也存在这个问题，长按动作释放后恢复了默认值而不是先前更改值

**Reply by verygoodlee** [reply]
inputevent应该没问题，按照这里的说法，`ignore #event: release`会自动恢复set, cycle, add, multiply这四种命令原始值
https://github.com/natural-harmonia-gropius/input-event?tab=readme-ov-file#how-does-the-auto-restore-on-release-works
也就是这个函数做的事，生成反向命令
https://github.com/dyphire/mpv-config/blob/32464e21f5d91c88d60ea35af27f1c511b7fd287/scripts/inputevent.lua#L177-L218

**Reply by WuYoue** [reply]
谢谢两位大佬的帮助，我通过该方法实现了长按倍速后恢复原速。😊💕

**Comment by dyphire** [comment]
在提交中 82181e8 更新`evafast.lua`脚本，优化倍速相关功能
